source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -939,9 +955,16 @@ namespace
                     {
                         // Write connections to upstream nodes.
                         RtOutput source = nodegraphInput.getConnection();
-                        RtNode sourceNode = source.getParent();
-                        input->setNodeName(sourceNode.getName());
-                        if (sourceNode.numOutputs() > 1)
+                        RtPrim sourcePrim = source.getParent();
```

Comment:
Fix save so that either "nodename" or "nodegraph" is written depending on the upstream connection.